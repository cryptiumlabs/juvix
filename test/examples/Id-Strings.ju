( b : Bah ->  c : B -o Foo) -> Foo a b -> a : Bah a c -o ( HAHAHHA -> foo )