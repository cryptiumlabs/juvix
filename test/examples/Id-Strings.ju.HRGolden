NoHeader
    [1m[96m[[0m ModuleOpen
        [1m[93m([0m Open [35m([0m [1m[97m"[0m[1m[94mPrelude[0m[1m[97m"[0m :| [36m[[0m[36m][0m [35m)[0m [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94mstring[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs = [33m[[0m[33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Primitive
                        [1m[35m([0m Prim
                            [1m[36m([0m [1m[97m"[0m[1m[94mMichelson[0m[1m[97m"[0m :| [1m[33m[[0m [1m[97m"[0m[1m[94mstring[0m[1m[97m"[0m [1m[33m][0m [1m[36m)[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94mpair[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs = [33m[[0m[33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Primitive
                        [1m[35m([0m Prim
                            [1m[36m([0m [1m[97m"[0m[1m[94mMichelson[0m[1m[97m"[0m :| [1m[33m[[0m [1m[97m"[0m[1m[94mpair[0m[1m[97m"[0m [1m[33m][0m [1m[36m)[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94munit[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs = [33m[[0m[33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Primitive
                        [1m[35m([0m Prim
                            [1m[36m([0m [1m[97m"[0m[1m[94mParam[0m[1m[97m"[0m :| [1m[33m[[0m [1m[97m"[0m[1m[94munit[0m[1m[97m"[0m [1m[33m][0m [1m[36m)[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs = [33m[[0m[33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Name [1m[35m([0m [1m[97m"[0m[1m[94munit[0m[1m[97m"[0m :| [1m[36m[[0m[1m[36m][0m [1m[35m)[0m [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Signature
        [1m[93m([0m Sig
            [35m{[0m signatureName = [1m[97m"[0m[1m[94mdefault[0m[1m[97m"[0m
            [35m,[0m signatureUsage = Nothing
            [35m,[0m signatureArrowType = Infix
                [36m([0m Inf
                    [33m{[0m infixLeft = Name [1m[35m([0m [1m[97m"[0m[1m[94mstring[0m[1m[97m"[0m :| [1m[36m[[0m[1m[36m][0m [1m[35m)[0m
                    [33m,[0m infixOp = [1m[97m"[0m[1m[94m->[0m[1m[97m"[0m :| [1m[35m[[0m[1m[35m][0m
                    [33m,[0m infixRight = Infix
                        [1m[35m([0m Inf
                            [1m[36m{[0m infixLeft = Name [1m[33m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m,[0m infixOp = [1m[97m"[0m[1m[94m->[0m[1m[97m"[0m :| [1m[33m[[0m[1m[33m][0m
                            [1m[36m,[0m infixRight = Name [1m[33m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m}[0m
                [36m)[0m
            [35m,[0m signatureConstraints = [36m[[0m[36m][0m
            [35m}[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94mdefault[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs =
                    [33m[[0m ConcreteA
                        [1m[35m([0m MatchLogic
                            [1m[36m{[0m matchLogicContents = MatchName [1m[97m"[0m[1m[94mpara[0m[1m[97m"[0m
                            [1m[36m,[0m matchLogicNamed = Nothing
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m,[0m ConcreteA
                        [1m[35m([0m MatchLogic
                            [1m[36m{[0m matchLogicContents = MatchName [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m
                            [1m[36m,[0m matchLogicNamed = Nothing
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Application
                        [1m[35m([0m App
                            [1m[36m{[0m applicationName = Name [1m[33m([0m [1m[97m"[0m[1m[94mpair[0m[1m[97m"[0m :| [95m[[0m[95m][0m [1m[33m)[0m
                            [1m[36m,[0m applicationArgs = List [1m[33m([0m ListLit [95m[[0m[95m][0m [1m[33m)[0m :|
                                [1m[33m[[0m Name [95m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [96m[[0m[96m][0m [95m)[0m [1m[33m][0m
                            [1m[36m}[0m
                        [1m[35m)[0m
                    [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m,[0m Signature
        [1m[93m([0m Sig
            [35m{[0m signatureName = [1m[97m"[0m[1m[94minit-storage[0m[1m[97m"[0m
            [35m,[0m signatureUsage = Nothing
            [35m,[0m signatureArrowType = Name [36m([0m [1m[97m"[0m[1m[94mstorage[0m[1m[97m"[0m :| [33m[[0m[33m][0m [36m)[0m
            [35m,[0m signatureConstraints = [36m[[0m[36m][0m
            [35m}[0m
        [1m[93m)[0m
    [1m[96m,[0m Function
        [1m[93m([0m Func
            [35m([0m Like
                [36m{[0m functionLikedName = [1m[97m"[0m[1m[94minit-storage[0m[1m[97m"[0m
                [36m,[0m functionLikeArgs = [33m[[0m[33m][0m
                [36m,[0m functionLikeBody = Body
                    [33m([0m Name [1m[35m([0m [1m[97m"[0m[1m[94munit[0m[1m[97m"[0m :| [1m[36m[[0m[1m[36m][0m [1m[35m)[0m [33m)[0m
                [36m}[0m
            [35m)[0m
        [1m[93m)[0m
    [1m[96m][0m