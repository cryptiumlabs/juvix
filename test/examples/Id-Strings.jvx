-- standard library functions

let string = "%prim.string"

let pair = "%pair"

-- Identity contract for strings. 

type storage = unit 

-- Default entry point
sig default : string -> storage -> storage
ï¿¼let default para storage = pair [] storage

-- Initialize storage to ().
sig initStorage : storage
let initStorage = ()
