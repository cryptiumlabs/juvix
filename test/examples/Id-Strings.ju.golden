
  [ Function'
      ( Func'
          ( Like'
              { functionLikedName = "string",
                functionLikeArgs = [],
                functionLikeBody = Body' (Primitive' (Prim' ("Michelson" :| ["string"]) ()) ()) (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "pair",
                functionLikeArgs = [],
                functionLikeBody = Body' (Primitive' (Prim' ("Michelson" :| ["pair"]) ()) ()) (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "unit",
                functionLikeArgs = [],
                functionLikeBody = Body' (Primitive' (Prim' ("Param" :| ["unit"]) ()) ()) (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "storage",
                functionLikeArgs = [],
                functionLikeBody = Body' (Name' ("unit" :| []) ()) (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Signature'
      ( Sig'
          { signatureName = "default",
            signatureUsage = Nothing,
            signatureArrowType =
              Infix'
                ( Inf'
                    { infixLeft = Name' ("string" :| []) (),
                      infixOp = "->" :| [],
                      infixRight =
                        Infix'
                          ( Inf'
                              { infixLeft = Name' ("storage" :| []) (),
                                infixOp = "->" :| [],
                                infixRight = Name' ("storage" :| []) (),
                                annInf = ()
                              }
                          )
                          (),
                      annInf = ()
                    }
                )
                (),
            signatureConstraints = [],
            annSig = ()
          }
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "default",
                functionLikeArgs =
                  [ ConcreteA'
                      ( MatchLogic'
                          { matchLogicContents = MatchName' "para" (),
                            matchLogicNamed = Nothing,
                            annMatchLogic = ()
                          }
                      )
                      (),
                    ConcreteA'
                      ( MatchLogic'
                          { matchLogicContents = MatchName' "storage" (),
                            matchLogicNamed = Nothing,
                            annMatchLogic = ()
                          }
                      )
                      ()
                  ],
                functionLikeBody =
                  Body'
                    ( Application'
                        ( App'
                            { applicationName = Name' ("pair" :| []) (),
                              applicationArgs = List' (ListLit' [] ()) () :| [Name' ("storage" :| []) ()],
                              annApp = ()
                            }
                        )
                        ()
                    )
                    (),
                annLike = ()
              }
          )
          ()
      )
      (),
    Signature'
      ( Sig'
          { signatureName = "init-storage",
            signatureUsage = Nothing,
            signatureArrowType = Name' ("storage" :| []) (),
            signatureConstraints = [],
            annSig = ()
          }
      )
      (),
    Function'
      ( Func'
          ( Like'
              { functionLikedName = "init-storage",
                functionLikeArgs = [],
                functionLikeBody = Body' (Name' ("unit" :| []) ()) (),
                annLike = ()
              }
          )
          ()
      )
      ()
  ]
